homeassistant/components/logbook/websocket_api.py

```diff
@@ -0,0 +1,266 @@
+    events = event_processor.get_events(start_day, end_day)
+    last_time = None
+    if events:
+        last_time = dt_util.utc_from_timestamp(events[-1]["when"])
+    result = formatter(msg_id, events)
+    return JSON_DUMP(result), last_time
+
+
+async def async_stream_events(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg_id: int,
+    stream_queue: asyncio.Queue[Event],
+    event_processor: EventProcessor,
+    last_time_from_db: dt,
+) -> None:
+    """Stream events from the queue."""
+    instance = get_instance(hass)
+    # Fetch any events from the database that have
+    # not been committed since the original fetch
+    await asyncio.sleep(instance.commit_interval)
+    message, last_time = await instance.async_add_executor_job(
+        _ws_formatted_get_events,
+        msg_id,
+        last_time_from_db,
+        dt_util.utcnow(),
+        messages.event_message,
+        event_processor,
+    )
+    if last_time:
+        connection.send_message(message)
+    else:
+        last_time = last_time_from_db
+
+    event_processor.switch_to_live()
+
+    while True:
+        events: list[Event] = [await stream_queue.get()]
+        # If the event is older than the last db
+        # event we already sent it so we skip it.
+        if events[0].time_fired <= last_time:
+            continue
+        await asyncio.sleep(EVENT_COALESCE_TIME)  # try to group events
+        while True:
+            try:
+                events.append(stream_queue.get_nowait())
+            except asyncio.QueueEmpty:
+                break
+
+        logbook_events = event_processor.humanify(
+            row for row in (async_event_to_row(e) for e in events) if row is not None
+        )
+        if not logbook_events:
+            continue
+
+        connection.send_message(
+            JSON_DUMP(
+                messages.event_message(
+                    msg_id,
+                    logbook_events,
+                )
+            )
+        )
+        stream_queue.task_done()
+
+
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "logbook/event_stream",
+        vol.Required("start_time"): str,
+        vol.Optional("entity_ids"): [str],
+        vol.Optional("device_ids"): [str],
+    }
+)
+@websocket_api.async_response
+async def ws_event_stream(
+    hass: HomeAssistant, connection: websocket_api.ActiveConnection, msg: dict
+) -> None:
+    """Handle logbook get events websocket command."""
+    start_time_str = msg["start_time"]
+    utc_now = dt_util.utcnow()
+
+    if start_time := dt_util.parse_datetime(start_time_str):
+        start_time = dt_util.as_utc(start_time)
+    else:
+        connection.send_error(msg["id"], "invalid_start_time", "Invalid start_time")
+        return
+
+    if start_time > utc_now:
+        connection.send_result(msg["id"], [])
+        return
+
+    device_ids = msg.get("device_ids")
+    entity_ids = msg.get("entity_ids")
```

Comment:
We should filter out entities that have UOM or state class here for the existing API

Reply:
195f5c1a8c8836c624d9353ed6de5f4836cc6462